File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-204311-76559/workdir/src/main/java/customerimporter/DNI.java
package customerimporter;

import java.util.function.Supplier;

public class DNI extends Identification {

    public static final String TYPE = "D";

    private static final int MINIMUM_NUMBER = 1;
    private static final int MAXIMUM_NUMBER = 99999999;

    private final int dniNumber;

    // error messages

    public static String invalidNumberErrorDescription() {
        return "Invalid DNI number";
    }

    // initialization

    DNI(String anIdentificationNumber) {
        super(anIdentificationNumber);
        dniNumber = Validation.assertIsNumberBetween(
            anIdentificationNumber, MINIMUM_NUMBER, MAXIMUM_NUMBER, invalidNumberErrorDescription());
    }

    // identification

    public String type() {
        return TYPE;
    }

    // testing

    public boolean isCUIT() {
        return false;
    }

    public boolean isDNI() {
        return true;
    }

    // numbers

    public Object cuitNumberIfNone(Supplier<Object> aNoneBlock) {
        return aNoneBlock.get();
    }

    public Object dniNumberIfNone(Supplier<Object> aNoneBlock) {
        return dniNumber;
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-205812-95418/analysis/acceptance-project/src/customerimporter/dni.py
from customerimporter.identification import Identification


class Dni(Identification):

    # error messages

    @classmethod
    def invalid_number_error_description(cls):
        return "Invalid DNI number"

    # type

    @classmethod
    def identification_type(cls):
        return "D"

    # number - validation

    @classmethod
    def assert_is_valid_number(cls, an_identification_number):
        cls.assert_valid_number(an_identification_number.isdigit())
        dni_number = int(an_identification_number)
        cls.assert_valid_number(dni_number >= 1 and dni_number <= 99999999)

    # number

    def cuit_number_if_none(self, a_none_block):
        return a_none_block()

    def dni_number_if_none(self, a_none_block):
        return int(self._number)

    def is_cuit(self):
        return False

    def is_dni(self):
        return True
